BUILDING.md

```diff
@@ -0,0 +1,246 @@
+# Build from source code
+- [Debian (+Linux Mint/Ubuntu)](#building-cherrytree-on-ubuntu-linux-mintubuntu)
+- [Arch Linux (+Manjaro Linux)](#building-cherrytree-on-arch-linux-linux-mintubuntu)
+- [Gentoo](#building-cherrytree-on-gentoo)
+- [Fedora](#building-cherrytree-on-fedora)
+- [Opensuse](#building-cherrytree-on-opensuse)
+- [MacOs](#building-cherrytree-on-macos)
+- [Windows](#building-cherrytree-on-windows)
+
+
+## Build/Debug with Visual Studio Code on Linux
+https://code.visualstudio.com/docs/setup/linux
+required installation of Extension "C/C++"
+```sh
+cd cherrytree
+code .
+```
+Build with: Ctrl+Shift+B
+Debug with: F5
+
+## Build/Debug with Visual Studio Code using a container
+It is possible to use a container as a full-featured development environment from VS Code.
+This works on any operating system that supports Docker.
+
+1. Install the [system requirements](https://code.visualstudio.com/docs/remote/containers#_system-requirements).
+2. Open the project in VS Code.
+3. (optional) Edit `.devcontainer/devcontainer.json` to set your `DISPLAY` environment variable and/or edit other settings.
+   It is also possible to run the container on a remote Docker host, see the comment at the end.
+4. Run the *Remote-Containers: Open Folder in Container...* command.
+5. See previous section for Build and Debug instructions.
+
+## To build using the bundled spdlog and fmt libraries
+```sh
+./build.sh bundledspdfmt
+```
+
+## To create an AppImage bundle
+```sh
+./build.sh appimage
+```
+
+## To generate a backtrace for a crash bug report
+```sh
+./build.sh debug
+gdb ./build/cherrytree
+(gdb) r
+```
+...after reproducing the crash
+```sh
+(gdb) bt
+```
+
+## Building Cherrytree on Ubuntu (+Linux Mint/Ubuntu)
```

Comment:
This should be changed to "... on Debian (+Ubuntu/Linux Mint)" as Ubuntu is based on Debian, and Linux Mint either on Ubuntu or Debian